src/rtree/trait.rs

```diff
@@ -160,15 +201,43 @@ pub trait RTreeIndex<N: IndexableNum>: Sized {
         y: N,
         max_results: Option<usize>,
         max_distance: Option<N>,
+    ) -> Vec<u32> {
+        // Use simple squared distance for backward compatibility
+        struct SimpleSquaredDistance;
+        impl<N: IndexableNum> SimpleDistanceMetric<N> for SimpleSquaredDistance {
+            fn distance(&self, x1: N, y1: N, x2: N, y2: N) -> N {
+                let dx = x2 - x1;
+                let dy = y2 - y1;
+                dx * dx + dy * dy
+            }
+            fn distance_to_bbox(&self, x: N, y: N, min_x: N, min_y: N, max_x: N, max_y: N) -> N {
+                let dx = axis_dist(x, min_x, max_x);
+                let dy = axis_dist(y, min_y, max_y);
+                dx * dx + dy * dy
+            }
+        }
```

Comment:
Can we move this to the top-level below the trait definition? It seems that I can't push to your branch